src/radio.rs

```diff
@@ -161,23 +159,62 @@ impl Phy {
             state: State::Idle,
         };
 
-        crate::crit_sect::locked(|cs| {
-            let prev_isr_data = ISR_DATA.borrow(cs).replace(Some(isr_data));
-            assert!(prev_isr_data.is_none());
-        });
+        let isr_data = Mutex::new(RefCell::new(isr_data));
 
-        Self {
-            isr_data: &ISR_DATA,
-        }
+        Self { isr_data }
+    }
+
+    fn get_global_instance_raw() -> &'static OnceCell<Phy> {
+        static PHY: OnceCell<Phy> = OnceCell::new();
+        &PHY
+    }
+
+    /// Set this Phy as global instance.
+    ///
+    /// Panics if global instance was already set.
+    pub fn set_as_global_instance(self) -> &'static Self {
```

Comment:
This function doesn't seem to need to be public
```suggestion
    fn set_as_global_instance(self) -> &'static Self {
```